src/se/vidstige/jadb/managers/Bash.java

```diff
@@ -1,14 +1,17 @@
 package se.vidstige.jadb.managers;
 
+import java.nio.charset.StandardCharsets;
+
 public class Bash {
     private Bash() {
         throw new IllegalStateException("Utility class");
     }
 
     public static String quote(String s) {
-        // Check that s contains no whitespace
-        if (s.matches("\\S+")) {
-            return s;
+        for (byte b : s.getBytes(StandardCharsets.UTF_8)) {
+            if (b < 32 || b > 126) {
+                throw new IllegalArgumentException("Illegal character in argument. Only use UTF-8 characters with indexes between 32 and 126");
```

Comment:
This seems incorrect. Try with "⾀" for example.

Reply:
Ah, yes, but this function is called `Bash.quote`. So filtering in this function is not a good idea. Bash can handle both ⾀ and å, ä, ö, etc.